service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/credential/mfa/shiro/MfaOptionServiceImpl.java

```diff
@@ -152,7 +155,7 @@ public MfaOption create(MfaOptionCreator mfaOptionCreator) throws KapuaException
 
             // generating base64 QR code image
             Account account = KapuaSecurityUtils.doPrivileged(() -> accountService.find(finalMfaOptionCreator.getScopeId()));
-            mfaOption.setQRCodeImage(generateQRCode(account.getName(), user.getName(), fullKey));
+            mfaOption.setQRCodeImage(generateQRCode(account.getOrganization().getName(), account.getName(), user.getName(), fullKey));
```

Comment:
Have you checked for NPEs if organization or organization name is `null`?

Reply:
The `create` method of the `AccountServiceImpl` performs the following check:  `ArgumentValidator.notEmptyOrNull(accountCreator.getOrganizationName(), "accountCreator.organizationName");`, thus the organization name is always null.